test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -565,15 +571,24 @@ private void ExecuteTFTransformMNISTConvTrainingTest(bool shuffle, int? shuffleS
                     }))
                     .Append(mlContext.Transforms.Concatenate("Features", "Prediction"))
                     .AppendCacheCheckpoint(mlContext)
-                    .Append(mlContext.MulticlassClassification.Trainers.LightGbm("Label", "Features"));
+                    .Append(mlContext.MulticlassClassification.Trainers.LightGbm(new LightGBM.Options() {
+                        LabelColumn="Label", 
+                        FeatureColumn="Features", 
+                        Seed=1, 
+                        NThread=1, 
+                        NumBoostRound=1}));
 
                 var trainedModel = pipe.Fit(preprocessedTrainData);
                 var predicted = trainedModel.Transform(preprocessedTestData);
                 var metrics = mlContext.MulticlassClassification.Evaluate(predicted);
 
                 // First group of checks. They check if the overall prediction quality is ok using a test set.
-                Assert.InRange(metrics.AccuracyMicro, expectedMicroAccuracy - .01, expectedMicroAccuracy + .01);
-                Assert.InRange(metrics.AccuracyMacro, expectedMacroAccruacy - .01, expectedMicroAccuracy + .01);
+                bool inRange = expectedValues.Any(exp =>
```

Comment:
Why not use `Assert.Equal` with a proper tolerance (e.g., 2) or `Assert.InRange`? They automatically print out error information.

Reply:
I wanted to use Assert.InRange, however this is given a list of expected ranges that should match (i.e. micro and macro should match one of those ranges). Because of this, Assert.InRange will not work because it will fail on the first range that doesnt match. 

---
In reply to: [254510687](https://github.com/dotnet/machinelearning/pull/2448#discussion_r254510687) [](ancestors = 254510687)